src/components/util/validation-functions.js

```diff
@@ -61,6 +88,61 @@ export function validateField(value, toValidate) {
     return makeErrorRecord(null);
 }
 
+/**
+ * Checks if the provided reactive capabilty curve is valid. Returns a list of
+ * errors if any, or an empty array otherwise.
+ * @param reactiveCapabilityCurve an array of reactive capability curve points of
+ * this format : [{ p: '', qminP: '', qmaxP: '' }, { p: '', qminP: '', qmaxP: '' }]
+ * @returns An array of error messages. If there is no error, returns an empty array.
+ */
+export function checkReactiveCapabilityCurve(reactiveCapabilityCurve) {
+    let errorMessages = [];
+
+    // At least four points must be set
+    if (reactiveCapabilityCurve.length < 2) {
+        errorMessages.push('ReactiveCapabilityCurveCreationErrorMissingPoints');
+    }
+
+    // Each P must be a unique valid number
+    const everyValidP = reactiveCapabilityCurve
+        .map((element) =>
+            validateValueIsANumber(element.p) ? element.p : null
+        )
+        .filter((p) => p !== null);
+    const setOfPs = [...new Set(everyValidP)];
```

Comment:
why not `const setOfPs = new Set(everyValidP);` ?

Reply:
Because I'd lose the ".length" function, and I want it so it is clearer in the code.